app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-ecommerce-tracker.js

```diff
@@ -97,7 +97,7 @@
         })
 
         ecommerceObject.event_data = {
-          external: GOVUK.analyticsGa4.analyticsModules.Ga4LinkTracker.isExternalLink(searchResult.getAttribute('data-ecommerce-path')) ? 'true' : 'false'
+          external: GOVUK.analyticsGa4.core.trackFunctions.isExternalLink(searchResult.getAttribute('data-ecommerce-path', [])) ? 'true' : 'false'
```

Comment:
You will need to make sure your internal domains array always has the current hostname in. With your current code, relative links and anchor links are still classed as internal, but full links such as `https://gov.uk/` will be tracked as external.

It might be worth adding tests to `ga4-link-tracker` with `https://gov.uk/` links and external links.

Reply:
I've refactored this but not sure I have enough tests in place, can you check and advise? Thanks.